### PR TITLE
Make sure index rule ID is set by

### PR DESCRIPTION
**Pull request by hanahmily**
* Add a default value based on 32 bits CRC to the ID
  * Check the ID on getting and listing index rules

Fixed https://github.com/apache/skywalking/issues/8846
Signed-off-by: Gao Hongtao <hanahmily@gmail.com>

### CODE REVIEW

File: banyand/metadata/schema/index.go [hunk 1]
```diff
@@ -19,16 +19,19 @@ package schema
 
 import (
 	"context"
+	"hash/crc32"
 
 	"google.golang.org/protobuf/proto"
 
 	commonv1 "github.com/apache/skywalking-banyandb/api/proto/banyandb/common/v1"
 	databasev1 "github.com/apache/skywalking-banyandb/api/proto/banyandb/database/v1"
+	"github.com/pkg/errors"
```
Comment (lujiajing1126) [inline banyand/metadata/schema/index.go:28]: the location of this import line is wrong

File: banyand/metadata/schema/index.go [hunk 2]
```diff
@@ -93,6 +96,9 @@ func (e *etcdSchemaRegistry) GetIndexRule(ctx context.Context, metadata *commonv
 	if err := e.get(ctx, formatIndexRuleKey(metadata), &entity); err != nil {
 		return nil, err
 	}
+	if entity.Metadata.Id == 0 {
+		return nil, errors.WithMessagef(ErrEmptyIndexRuleID, "index rule: %v", metadata)
```
Comment (lujiajing1126) [inline banyand/metadata/schema/index.go:100]: Shall we return grpc status code since it it API related

File: banyand/metadata/schema/index.go [hunk 3]
```diff
@@ -108,12 +114,21 @@ func (e *etcdSchemaRegistry) ListIndexRule(ctx context.Context, opt ListOpt) ([]
 	}
 	entities := make([]*databasev1.IndexRule, 0, len(messages))
 	for _, message := range messages {
-		entities = append(entities, message.(*databasev1.IndexRule))
+		entity := message.(*databasev1.IndexRule)
+		if entity.Metadata.Id == 0 {
+			return nil, errors.WithMessagef(ErrEmptyIndexRuleID, "index rule: %v", entity.Metadata)
```
Comment (lujiajing1126) [inline banyand/metadata/schema/index.go:119]: same here

**Review by lujiajing1126 — CHANGES_REQUESTED** [review]

File: banyand/metadata/schema/index.go (hunk 1 above)
Comment (hanahmily) [reply banyand/metadata/schema/index.go]: forget to run `make pre-push`

File: banyand/metadata/schema/index.go (hunk 2 above)
Comment (hanahmily) [reply banyand/metadata/schema/index.go:100]: Make sense. I will introduce `DataLoss` to indicate this kind of data corruption.

**Review by lujiajing1126 — APPROVED** [review]
LGTM

**Pull request merged**